docs/developer/pull-request.md

```diff
@@ -0,0 +1,107 @@
+- Use your best judgement when resolving comments. Simple, unambiguous comments should be resolved by the submitter,
+  this prioritizes speed and efficacy. For larger changes, for example, when updating logic and design, the resolution
+  should be left for the reviewer’s approval or continued discussion.
+- If a discussion grows too long - more than a handful (3 - 5) back and forth responses, consider moving the discussion
+  to a more real-time platform: Slack, Zoom, Phone, In-person. Do this to clear misunderstandings or whiteboard
+  improvements. Whenever the conversation has moved to a different platform and found a conclusion, the decision and
+  resolution MUST be updated in the review, for instance,
+
+  ```text
+  Comment:
+
+  Spoke offline, decided to move A to package B. Rewrite tests and API docs.
+  ```
+
+- When pushing code review changes, it's important to provide clear information to the reviewer. Here are a couple of
+  guidelines to follow:
+    - Commit each code review change individually.
+    - Use descriptive commit messages that accurately describe the change made, making it easy for the reviewer to
+      understand the purpose of the change.
+- When approved, and all review comments have been resolved; squash commits to a single commit. Follow
+  the [Commit Message Format](#commit-message-format) guidelines when writing the final commit.
+
+## Reviewer Guidelines
+
+- As a reviewer, you bear the ultimate responsibility for the resolution of the comment. If not resolved, please
+  follow-up in a timely manner with acceptance of the solution and explanation.
+- Do your best to review in a timely manner. However, code reviews are time-consuming, maximize their benefit by
+  focusing on what’s highest value. This code review pyramid outlines a reasonable shorthand:
+  ![Code Review Pyramid](code-review-pyramid.jpeg)
+- Always review for: design, API semantics, [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) and
+  maintainability practices, bugs and quality, efficiency and correctness.
+- Code review checklist:
+    - Do any changes need to made in documentation?
+        - Do public docs need updating?
+        - Have internal workflow and informational docs been impacted?
+    - Are there unit tests for the change?
+        - If the change is a bug fix, has a unit test been added or modified that catches the bug? All bug fixes should
+          be reproduced with a unit test before submitting any code. Then the code should be fixed so that the unit test
+          passes.
+        - If the change is a feature or new functionality. The expectation is to include a reasonable set of unit tests
+          which exercise positive and negative cases.
+    - Is this change backwards compatible? Is it causing breaking behavior?
+    - Are there any [Comment Tags](#comment-tags) in the change?
+
+## Commit Message Format
+
+The commit message for your final commit should use a consistent format:
+
+```text
```

Comment:
we don't really follow that.

Can we have something a bit simpler?

```
One line summary of the change>

Summary of the changes.

Reference to the GH issue
```

Reply:
I can remove the testing portion, but I do think we should follow the problem/solution format for bugs.